fvm/evm/emulator/state/stateDB.go

```diff
@@ -308,12 +308,44 @@ func (db *StateDB) SetTransientState(addr gethCommon.Address, key, value gethCom
 
 // AddressInAccessList checks if an address is in the access list
 func (db *StateDB) AddressInAccessList(addr gethCommon.Address) bool {
-	return db.latestView().AddressInAccessList(addr)
+	// For each static call / call / delegate call, the EVM will create
+	// a snapshot, so that it can revert to it in case of execution errors,
+	// such as out of gas etc, using `Snapshot` & `RevertToSnapshot`.
+	// This can create a long list of views, in the order of 4K for certain
+	// large transactions. To avoid performance issues with deep stacks,
+	// we use a plain for-loop, instead of checking parents, as DeltaView did.
+	// We take advantage of the fact that addresses are added in AccessList
+	// in the early steps of tx execution, so we return early.
+	for _, view := range db.views {
```

Comment:
The previous implementation starts from `latestView` to traverse backwards. The iteration range of the views is pinned to be `[views[0], latestView]`, now we are changing to iterate through `db.views`, since `db.views` can potentially be updated by `Snapshot` call, is it safe to assume the `db.views` won't be updated during this call?

What about we pin the end view index first, and then iterate:
```suggestion
	end := len(db.views)
	for i := 0; i < end; i++ {
		view := db.views[i]
```